std/array.asm

```diff
@@ -14,4 +14,10 @@ let fold = |arr, initial, folder| std::utils::fold(len(arr), |i| arr[i], initial
 
 /// Returns the sum of the array elements.
 /// This actually also works on field elements, so the type is currently too restrictive.
-let sum: int[] -> int = |arr| fold(arr, 0, |a, b| a + b);
\ No newline at end of file
+let sum: int[] -> int = |arr| fold(arr, 0, |a, b| a + b);
+
+/// Zips two arrays
+let zip2 = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));
+
+/// Zip three arrays
+let zip3 = |array1, array2, array3, fn| new(len(array1), |i| fn(array1[i], array2[i], array3[i]));
```

Comment:
We could also have one zip function instead of one for each number of arrays. It would take an array of arrays.
the downside is that that `fn` would then receive an array and index its elements, which is less readable. I guess at some point we can have tuple unpacking and fix that.

Reply:
Can you use `zip2` inside `zip3`?

Reply:
That's the best I can come up with and I don't think it's an improvement.

```suggestion
/// Zips two arrays
let zip2 = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));

/// Zip three arrays
let zip3 = |array1, array2, array3, fn| zip2(
   zip2(array1, array2, |array1_i, array2_i| [array1_i, array2_i]),
   array3,
   |array12_i, array3_i| fn(array12_i[0], array12_i[1], array3_i)
);
```